File: library/src/main/java/com/vorlonsoft/android/rate/DialogType.java
/*
 * Copyright 2018 Vorlonsoft LLC
 *
 * Licensed under The MIT License (MIT)
 */

package com.vorlonsoft.android.rate;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;

import androidx.annotation.IntDef;

import static com.vorlonsoft.android.rate.Constants.Utils.UTILITY_CLASS_MESSAGE;
import static java.lang.annotation.RetentionPolicy.SOURCE;

/**
 * <p>DialogType Class - dialog type class of the AndroidRate library.</p>
 *
 * @since    1.2.1
 * @version  1.2.1
 * @author   Alexander Savin
 */

@SuppressWarnings({"WeakerAccess", "unused"})
public class DialogType {
    /** <p>Apple Rate Dialog.</p> */
    public static final int APPLE = 0;
    /** <p>Classic Rate Dialog.</p> */
    public static final int CLASSIC = 1;
    /** <p>Modern Rate Dialog.</p> */
    public static final int MODERN = 2;

    private DialogType() {
        throw new UnsupportedOperationException("DialogType" + UTILITY_CLASS_MESSAGE);
    }

    /**
     * <p>Annotates element of integer type.</p>
     * <p>Annotated element, represents a logical type and its value should be one of the following
     * constants: APPLE, CLASSIC, MODERN.</p>
     */
    @Documented
    @Retention(SOURCE)
    @IntDef({
            APPLE,
            CLASSIC,
            MODERN
    })
    public @interface AnyDialogType {
    }
}

File: library/src/main/java/com/vorlonsoft/android/rate/DialogType.kt
/*
 * Copyright 2018 Vorlonsoft LLC
 *
 * Licensed under The MIT License (MIT)
 */

package com.vorlonsoft.android.rate

import androidx.annotation.IntDef

/**
 * DialogType Class - the dialog types class of the AndroidRate library.
 *
 * Contains dialog types constants and [AnyDialogType] annotation.
 *
 * @constructor Don't create an instance of this class. Use its members directly.
 * @since       1.2.1
 * @version     1.2.1
 * @author      Alexander Savin
 */
open class DialogType {
    /**
     * Denotes that the annotated element of the primitive type int represents a logical type and
     * that its value should be one of the following constants: [APPLE], [CLASSIC], [MODERN].
     *
     * @since       1.2.1
     * @version     1.2.1
     * @author      Alexander Savin
     */
    @MustBeDocumented
    @Retention(AnnotationRetention.SOURCE)
    @IntDef(APPLE, CLASSIC, MODERN)
    annotation class AnyDialogType

    /** Contains constants for dialog types. */
    companion object {
        /** Apple Rate Dialog. */
        const val APPLE: Int = 0
        /** Classic Rate Dialog. */
        const val CLASSIC: Int = 1
        /** Modern Rate Dialog. */
        const val MODERN: Int = 2
    }
}